src/types.js

```diff
@@ -99,6 +95,13 @@ export type DroppableDimension = {|
     // the area in which content presses up against
     withMarginAndPadding: DimensionFragment,
   |},
+  container: {|
```

Comment:
I am a little worried that it might not be clear that the 'container' might actually be itself. Perhaps the label `closestScrollable` is clearer.

Reply:
yeah it's a tough one because sometimes the droppable won't have to consider scrolling at all - it won't have internal scrolling and it won't be within a scroll parent so its container will be itself.

essentially what this dimension represents is the maximum possible extent of the droppable's visible bounds. this could be either:
* its own dimensions (when it has no internal scrolling and isn't inside a scrolling parent)
* its max-height/max-width (when it has internal scrolling)
* the dimension of its scroll parent (when it's inside a scrolling parent). note: in this case its container dimension may actually be larger than its own dimension (imagine multiple droppables inside a shared scroll parent), but this scroll parent still constrains its potential visible area.

in light of this, 'container' seemed like the most appropriately generic terminology... 🤔  do you have any other suggestions? totally open to changing this if something else makes more sense to you :)

Reply:
more diagrams 🙌
![container_dimensions](https://user-images.githubusercontent.com/4995433/30674910-90bc921e-9ec0-11e7-893e-e00fc0cc08bd.png)

Reply:
I'll leave it to you about name - but definitely add a good comment explaining what it is. The diagrams look great - be sure to include them in the release notes if it makes sense :)

What do you think about making `page` > `withMargin` and `withoutMargin` (for consistency)

Reply:
🤔 we can add a withoutMargin property for object shape consistency if you want to, but I don't think we'll ever care about a container's withMargin dimensions because we only need to know its content box.

I'll do something like this for consistency:
```javascript
container: {|
  scroll: {|
    initial: Position,
    current: Position,
  |},
  page: {|
    withoutMargin: DimensionFragment,
  |},
|}
```

Reply:
perfect